ui-components/src/authentication/AuthContext.ts

```diff
@@ -220,6 +220,14 @@ export const AuthProvider: FunctionComponent = ({ children }) => {
     }
   }, [apiUrl, storageType])
 
+  const enforcePreferences = (profile: User) => {
+    if (!profile.preferences) {
+      profile.preferences = {
+        favoriteIds: [],
+      }
+    }
+  }
+
```

Comment:
@YazeedLoonat ,
Rather than utilize this client side, would it be better to specify this in the `user.dto` preferences definition server side?

Reply:
Hey @seanmalbert I don't know if thats the right call since a user doesn't have to have preferences the only reason this needs preferences is the way  the front end is asking for it. I spun up: https://github.com/CityOfDetroit/bloom/pull/1266

which removes AuthContext needing to know that we need favoriteIds array, and instead creates a pathway for us to update the profile. This then allows the favorite button themselves to add onto the preferences as they need it when we go to favorite a listing. I think this flows a bit better, but lemme know what you think about that pr/not updating the backend for this

Reply:
Thanks for putting together the other PR and after looking at that, I agree that it is the preferred option.